Themis/ViewModels/Assessment/File Upload/FileRendererViewModel.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  FileRendererViewModel.swift
+//  Themis
+//
+//  Created by Tarlan Ismayilsoy on 11.09.23.
+//
+
+import Foundation
+import SharedModels
+import UserStore
+import Common
+
+class FileRendererViewModel: ExerciseRendererViewModel {
+    @Published var remoteFileURL: URL?
+    @Published var localFileURL: URL?
+    @Published var fileExtension: FileUploadExerciseFileExtension?
+    @Published var isSetupComplete = false
+    @Published var canDirectlyRenderFile = false
+    
+    var fileDownloadService = FileDownloadService()
+    
+    /// Sets this VM up based on the given submission
+    @MainActor
+    func setup(basedOn submission: BaseSubmission? = nil) async {
+        guard let fileUploadSubmission = submission as? FileUploadSubmission,
+              let baseUrl = UserSession.shared.institution?.baseURL?.absoluteString,
+              let filePath = fileUploadSubmission.filePath?.dropFirst(),
```

Comment:
This line requires at least [artemis-ios-core-modules](https://github.com/ls1intum/artemis-ios-core-modules) version 3.4.0! 
This should be adapted in the project file.